todo-app/src/Todo.js

```diff
@@ -0,0 +1,129 @@
+import React from 'react';
+import Description from './Description';
+import Time from './Time';
+import Button from './button';
+import TextField from './TextField';
+
+export default class Todo extends React.Component{
+   
+    state = {
+        todoText: '',
+        deadline: '',
+    };
+
+    componentWillReceiveProps(newProps) {
+        if (newProps.isEditing && !this.props.isEditing) {
+          this.setState({
+              todoText: newProps.text,
+              deadline: newProps.deadline,});
+        }
+    };
+
+    handleTextChange = event => {
+        this.setState({todoText: event.target.value});
+    };
+
+    handleDeadlineChange = event => {
+        this.setState({deadline: event.target.value});
+    };
+
+    handleSaveClick = () => {
+        const {todoText, deadline } = this.state;
+        this.props.onSaveClick(todoText);
+        this.setState({todoText: ''});
+        this.props.onSaveDateClick(deadline);
```

Comment:
Why do you have two `onSave...` methods? If you always call them together, why not have one single `onSave` method which passes an entire `todo` object?